practice_app/services/getLocationInfo.spec.js

```diff
@@ -1,4 +1,174 @@
+                        symbol: 'a',
+                        code: 'b',
+                        name: 'c',
+                    },
+                ],
+                languages: {
+                    tur: 'Turkish',
+                    en: 'English',
+                },
+            },
+        }
+
+        const expected = {
+            symbol: 'a',
+            code: 'b',
+            name: 'c',
+        }
+
+        const output = getCurrency(input)
+        expect(output).toEqual(expected)
+    })
+
+    test('returns nothing when currencies have 0 elements', () => {
+        const input = {
+            country: 'Turkey',
+            country_module: {
+                currencies: [],
+                languages: {
+                    tur: 'Turkish',
+                    en: 'English',
+                },
+            },
+        }
+
+        const output = getCurrency(input)
+        expect(output).toEqual(undefined)
+    })
+
+    test('returns nothing when currencies does not exist', () => {
+        const input = {
+            country: 'Turkey',
+            country_module: {
+                languages: {
+                    tur: 'Turkish',
+                    en: 'English',
+                },
+            },
+        }
+
+        const output = getCurrency(input)
+        expect(output).toEqual(undefined)
+    })
+
+    test('returns nothing when country_module does not exist', () => {
+        const input = {
+            country: 'Turkey',
+        }
+
+        const output = getCurrency(input)
+        expect(output).toEqual(undefined)
+    })
+})
+
+describe('getLanguage', () => {
+    test('returns expected language on valid input', () => {
+        const input = {
+            country: 'Turkey',
+            country_module: {
+                currencies: [
+                    {
+                        symbol: 'a',
+                        code: 'b',
+                        name: 'c',
+                    },
+                ],
+                languages: {
+                    tur: 'Turkish',
+                    en: 'English',
+                },
+            },
+        }
+
+        const output = getLanguage(input)
+        expect(output).toStrictEqual(input.country_module.languages.tur)
+    })
+
+    test('returns nothing when country_module does not exist.', () => {
+        const input = {
+            country: 'Turkey',
+        }
+
+        const output = getLanguage(input)
+
+        expect(output).toBe(undefined)
+    })
+
+    test('returns nothin when country_module.languages doesnt exist.', () => {
+        const input = {
+            country: 'Turkey',
+            country_module: {
+                currencies: [
+                    {
+                        symbol: 'a',
+                        code: 'b',
+                        name: 'c',
+                    },
+                ],
+            },
+        }
+
+        const output = getLanguage(input)
+
+        expect(output).toBe(undefined)
```

Comment:
Sometimes you use .toBe(undefined) other times .toEqual(undefined). For the sake of consistency please use .toStrictEqual(undefined) everywhere

Reply:
fixed